app/ts/Workspace/WorkspaceContext.tsx

```diff
@@ -44,12 +49,32 @@ export function WorkspaceContextProviderComponent({
     Record<string, [number, number]>
   >({});
 
+  const [workspace, setWorkspace] = useState<IWorkspace>(
+    buildWorkspace(oldWorkspace, workspaceId)
+  );
+  const [ordering, setOrdering] = useState<IOrdering>();
+
+  useEffect(() => {
+    Axios.get(`/workspaces/${workspaceId}/ordering`)
+      .then((response: AxiosResponse<{ordering: IOrdering | {}}>) => {
+        const newOrdering: IOrdering | {} = response.data.ordering;
+        if (isOrdering(newOrdering)) {
+          setOrdering(newOrdering);
+          setWorkspace(buildWorkspace(oldWorkspace, workspaceId, newOrdering));
+        }
+      })
+      .catch(errorCallback);
+  }, []);
+
   useEffect(() => {
     if (scales && oldWorkspace) {
       setObservedRanges(calculateObservedRanges(scales, workspace));
     }
   }, [scales, oldWorkspace]);
-  const workspace: IWorkspace = buildWorkspace(oldWorkspace, workspaceId);
+
+  function isOrdering(ordering: {} | IOrdering): ordering is IOrdering {
+    return !_.isEqual(ordering, {});
```

Comment:
An object is of type IOrdering if it is not an empty object? I'm confused

Reply:
The backend sends an empty object if there is no ordering available for the workspace. It's an annoying special case :(

Reply:
OK, so this is a minimal implementation with that in the back of your head, but it reads weird; I'd prefer something more directly readable like `_.equals(['criteria', 'alternatives', 'dataSources'], _.keys(object))`

Reply:
How do you know the order of what keys returns? Leaving as is for now.